homeassistant/components/media_player/dlna_dmr.py

```diff
@@ -0,0 +1,717 @@
+# -*- coding: utf-8 -*-
+"""
+Support for DLNA DMR (Device Media Renderer).
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.dlna_dmr/
+"""
+
+import asyncio
+import functools
+import logging
+import re
+import urllib.parse
+from datetime import timedelta
+
+import aiohttp
+import async_timeout
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.http.view import (
+    request_handler_factory, HomeAssistantView)
+from homeassistant.components.media_player import (
+    SUPPORT_PLAY, SUPPORT_PAUSE, SUPPORT_STOP,
+    SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_SET,
+    SUPPORT_PLAY_MEDIA,
+    SUPPORT_PREVIOUS_TRACK, SUPPORT_NEXT_TRACK,
+    MediaPlayerDevice,
+    PLATFORM_SCHEMA)
+from homeassistant.const import (
+    EVENT_HOMEASSISTANT_STOP,
+    CONF_URL, CONF_NAME,
+    STATE_OFF, STATE_ON, STATE_IDLE, STATE_PLAYING, STATE_PAUSED)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+
+REQUIREMENTS = [
+    'async-upnp-client==0.10.1',
+    'python-didl-lite==1.0.1',
+]
+
+DEPENDENCIES = ['http']
+
+DEFAULT_NAME = 'DLNA Digital Media Renderer'
+
+CONF_UDN = 'udn'
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_URL): cv.string,
+    vol.Required(CONF_UDN): cv.string,
```

Comment:
Please do not make UDN entry a requirement as it can be read from the service desc file. It is painful enough to find out what URL you should be using, requiring users to parse the XML for UDN is a no-go in my eyes.

Reply:
Ok, will change.